File: DotNet/Node.Core/Biz/Interfaces/GetServices/IPreProcess.cs
using System;

using Node.Core.Biz.Manageable.Parameters;
using Node.Core.Biz.Objects;

namespace Node.Core.Biz.Interfaces.GetServices
{
    /// <summary>
    /// The Interface for GetServices PreProcess
    /// </summary>
    public interface IPreProcess
    {
        /// <summary>
        /// PreProcess for GetServices.
        /// </summary>
        /// <param name="token">The Token which is result of Authentication Operation.</param>
        /// <param name="serviceType">Specifies which type of service are to be used.</param>
        /// <param name="param">Operation related parameters</param>
        /// <returns>A list of Opeation Name</returns>
        void Execute(string token, string serviceType, PreParam param);
    }
}

File: EN Node for Java environment/Node/src/Node/Biz/Interfaces/GetServices/IPreProcess.java
package Node.Biz.Interfaces.GetServices;

import java.rmi.RemoteException;

import Node.Biz.Custom.PreParam;
/**
 * <p>This class create GetServices Pre Process Interface.</p>
 * <p>Company: enfoTech & Consulting, Inc.</p>
 * @author enfoTech
 * @version 2.0
 */

public interface IPreProcess {
	/**
	 * Execute
	 * @param token The authentication token
	 * @param serviceType The service type name
	 * @param param PreParam object
	 * @throws RemoteException
	 * @return PreParam object
	 */
  public PreParam Execute (String token, String serviceType, PreParam param) throws RemoteException;
}
